File: ios/Runner/AmapParam.swift
//
//  AmapParam.swift
//  Runner
//
//  Created by xionghao on 2020/3/7.
//  Copyright © 2020 The Chromium Authors. All rights reserved.
//

//enum MapType: Int {
//   case routeMap = 0
//   case addressDescriptionMap = 1
//}

class AmapParam: Codable {
    
    static let routeMap: Int = 0
    static let addressDescriptionMap: Int = 0
    
    let initialCenterPoint: Array<Double>
    let initialZoomLevel: CGFloat
    let enableMyLocation: Bool?
    let enableMyMarker: Bool?
    let mapType: Int
    let startAddressList: Array<AddressInfo>?
    let endAddressList: Array<AddressInfo>?
    
    init(
        initialCenterPoint: Array<Double>,
        initialZoomLevel: CGFloat,
        enableMyLocation: Bool? = false,
        enableMyMarker: Bool? = false,
        mapType: Int = routeMap,
        startAddressList: Array<AddressInfo>? = [],
        endAddressList: Array<AddressInfo>? = []
    ) {
        self.initialCenterPoint = initialCenterPoint
        self.initialZoomLevel = initialZoomLevel
        self.enableMyLocation = enableMyLocation
        self.enableMyMarker = enableMyMarker
        self.mapType = mapType
        self.startAddressList = startAddressList
        self.endAddressList = endAddressList
    }
}

struct AddressInfo: Codable {
    let geo: GeoPoint?
    let address: String?
}

struct GeoPoint: Codable {
    let lat: Double?
    let lng: Double?
}

File: android/app/src/main/kotlin/xh/zero/flutter_demo/plugins/amap/AmapParam.kt
package xh.zero.flutter_demo.plugins.amap

class AmapParam {
    val initialCenterPoint: ArrayList<Double>? = null
    val initialZoomLevel: Float? = null
    val enableMyLocation: Boolean = false
    val enableMyMarker: Boolean = false
    val mapType: Int? = ROUTE_MAP

    val startAddressList: ArrayList<AddressInfo>? = null
    val endAddressList: ArrayList<AddressInfo>? = null

    class AddressInfo {
        val geo: GeoPoint? = null
        val address: String? = null
    }

    class GeoPoint(val lat: Double, val lng: Double)

    companion object {
        const val ROUTE_MAP: Int = 0
        const val ADDRESS_DESCRIPTION_MAP: Int = 1
    }
}